card/src/main/java/com/adyen/checkout/card/util/AddressFormUtils.kt

```diff
@@ -0,0 +1,158 @@
+package com.adyen.checkout.card.util
+
+import com.adyen.checkout.card.AddressConfiguration
+import com.adyen.checkout.card.AddressFormUIState
+import com.adyen.checkout.card.AddressOutputData
+import com.adyen.checkout.card.api.model.AddressItem
+import com.adyen.checkout.card.ui.model.AddressListItem
+import com.adyen.checkout.components.base.AddressVisibility
+import com.adyen.checkout.components.model.payments.request.Address
+
+internal object AddressFormUtils {
+
+    /**
+     * Map a list of [AddressItem] to a list of [AddressListItem].
+     *
+     * @param list Input list.
+     * @param shouldSelectFirstItem Whether or not the first item should be marked as selected.
+     *
+     * @return Mapped list of [AddressListItem].
+     */
+    fun mapToListItem(list: List<AddressItem>, shouldSelectFirstItem: Boolean): List<AddressListItem> {
+        return list.map {
+            val isFirstItem = it.id == list.firstOrNull()?.id
+            AddressListItem(
+                name = it.name.orEmpty(),
+                code = it.id.orEmpty(),
+                selected = shouldSelectFirstItem && isFirstItem
+            )
+        }
+    }
+
+    /**
+     * Mark the item that matches the given code as selected in the given input list.
+     *
+     * @param list Input list of [AddressListItem].
+     * @param code Country or state code to be marked as selected.
+     *
+     * @return List of [AddressListItem] with the item in the list having given code marked as selected.
+     */
+    fun markAddressListItemSelected(list: List<AddressListItem>, code: String): List<AddressListItem> {
+        return list.map {
+            it.copy(selected = it.code == code)
+        }
+    }
+
+    /**
+     * Get visibility state of the address form.
+     *
+     * @param addressConfiguration Configuration object for address form.
+     * @param addressVisibility Visibility modifier for address form.
+     * @param isStoredCard Whether or not current form is for stored card or a new card.
+     *
+     * @return Visibility state of the address form.
+     */
+    fun getAddressFormUIState(
+        addressConfiguration: AddressConfiguration?,
+        addressVisibility: AddressVisibility,
+        isStoredCard: Boolean
+    ): AddressFormUIState {
+        return when {
+            isStoredCard -> AddressFormUIState.NONE
```

Comment:
Note just for consistency, in other places the way we distinguish between stored and not stored is to have a method in `CardDelegate` and make `NewCardDelegate` and `StoredCardDelegate` return different stuff.

So maybe here we could have:

```kotlin
// AddressFormUtils:
fun getAddressFormUIState(
        addressConfiguration: AddressConfiguration?,
        addressVisibility: AddressVisibility
    ): AddressFormUIState {
        return when {
            addressConfiguration != null -> AddressFormUIState.fromAddressConfiguration(addressConfiguration)
            else -> AddressFormUIState.fromAddressVisibility(addressVisibility)
        }
    }

// CardDelegate:
abstract fun getAddressFormUIState(
        addressConfiguration: AddressConfiguration?,
        addressVisibility: AddressVisibility
): AddressFormUIState

// NewCardDelegate
override fun getAddressFormUIState(
        addressConfiguration: AddressConfiguration?,
        addressVisibility: AddressVisibility
): AddressFormUIState {
    return AddressFormUtils.getAddressFormUIState(
        addressConfiguration,
        addressVisibility
    )
}

// StoredCardDelegate:
override fun getAddressFormUIState(
        addressConfiguration: AddressConfiguration?,
        addressVisibility: AddressVisibility
): AddressFormUIState {
    return AddressFormUIState.NONE
}
```

WDYT?